plugins/newsdownloader.koplugin/epubdownloadbackend.lua

```diff
@@ -191,6 +191,24 @@ local function getUrlContent(url, cookies, timeout, maxtime, redirectCount)
         logger.warn("request interrupted:", status or code)
         return false, code
     end
+    if code >= 400 and code < 500 then
+        logger.warn("socket error:", status or code)
+        if code == 400 then
+            return false, "Bad request - invalid URL or parameters"
```

Comment:
I have no idea what's going on here tbh. It seems incredibly redundant with `status`. I'll just do:

```suggestion
            return false, status or code
```